### PR TITLE
Making rabbitmq dependency optional on console

### PR DESCRIPTION
**Pull request by marwanehcine**
On this PR, we make  rabbitmq dependency optional on console

### CODE REVIEW

**Review by pmauduit — COMMENTED** [review]
Thanks Marwanne, it indeed looks more relevant than the previous one without mixing spring-boot components.

I am wondering what a ApplicationContextInitializer provides though, why not just sourcing the /etc/georchestra/default.properties directly from the xml file ?

**Comment by marwanehcine** [comment]
> Thanks Marwanne, it indeed looks more relevant than the previous one without mixing spring-boot components.
> 
> I am wondering what a ApplicationContextInitializer provides though, why not just sourcing the /etc/georchestra/default.properties directly from the xml file ?

The problem here is that <import>  in xml conf file for spring 3 are called before reading props file, so the only solution is to define listerner on web.xml to source default.props before <import>.  this is the only way it works!

File: console/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -14,6 +14,10 @@
         <param-name>org.eclipse.jetty.servlet.Default.dirAllowed</param-name>
         <param-value>false</param-value>
     </context-param>
+    <context-param>
+        <param-name>contextInitializerClasses</param-name>
+        <param-value>org.georchestra.console.events.DefaultPropsInitializer</param-value>
+    </context-param>
```
[the remark this answers is not in the document]
Comment (marwanehcine) [reply console/src/main/webapp/WEB-INF/web.xml:20]: Hello, good question, issue here is that "<import>" inside xml conf file (beans definition) is called too early so props files a not accessible yet, however using context-param, we load default.properties file even before call to <import> on this case.
I have checked other cases and it did not work only after user of context-param inside web.xml file

File: console/src/main/webapp/WEB-INF/spring/webmvc-config.xml [hunk 2]
```diff
@@ -324,12 +324,5 @@
...
 
-  <bean id="rabbitmqEventsSender" class="org.georchestra.console.events.RabbitmqEventsSender" depends-on="eventTemplate">
-    <property name="eventTemplate" ref="eventTemplate"/>
-  </bean>
-
-  <import resource="file:webapps/console/WEB-INF/spring/rabbit-listener-context.xml"/>
-
-  <import resource="file:webapps/console/WEB-INF/spring/rabbit-sender-context.xml"/>
-
+  <import resource="classpath:spring/rabbit-context-${enableRabbitmqEvents}.xml"/>
```
[the remark this answers is not in the document]
Comment (marwanehcine) [reply console/src/main/webapp/WEB-INF/spring/webmvc-config.xml:327]: code updated

File: console/src/main/java/org/georchestra/console/events/DefaultPropsInitializer.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.io.IOException;
+
+public class DefaultPropsInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+
+    @Override
+    public void initialize(ConfigurableApplicationContext applicationContext) {
+        try {
+            applicationContext.getEnvironment().getPropertySources().addFirst(
+                    new ResourcePropertySource(new FileSystemResource("/etc/georchestra/default.properties")));
```
[the remark this answers is not in the document]
Comment (marwanehcine) [reply console/src/main/java/org/georchestra/console/events/DefaultPropsInitializer.java:16]: code updated!

File: console/src/main/java/org/georchestra/console/events/DefaultPropsInitializer.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Override
+    public void initialize(ConfigurableApplicationContext applicationContext) {
+        try {
+            applicationContext.getEnvironment().getPropertySources()
+                    .addFirst(new ResourcePropertySource(new FileSystemResource(
+                            System.getProperty("georchestra.datadir", "etc/georchestra") + "/default.properties")));
+        } catch (IOException e) {
+            LOGGER.debug("Cannot find default props file");
+            LOGGER.error(e.getStackTrace());
```
Comment (pmauduit) [inline console/src/main/java/org/georchestra/console/events/DefaultPropsInitializer.java:23]: maybe LOGGER.error for the first one as well ?

Comment (pmauduit) [reply console/src/main/java/org/georchestra/console/events/DefaultPropsInitializer.java:23]: or even `LOGGER.error("Cannot find ...", e);`

File: console/src/main/webapp/WEB-INF/spring/webmvc-config.xml [hunk 5]
```diff
@@ -324,12 +324,5 @@
...
 
-  <bean id="rabbitmqEventsSender" class="org.georchestra.console.events.RabbitmqEventsSender" depends-on="eventTemplate">
-    <property name="eventTemplate" ref="eventTemplate"/>
-  </bean>
-
-  <import resource="file:webapps/console/WEB-INF/spring/rabbit-listener-context.xml"/>
-
-  <import resource="file:webapps/console/WEB-INF/spring/rabbit-sender-context.xml"/>
-
+  <import resource="classpath:spring/rabbit-context-${enableRabbitmqEvents:true}.xml"/>
```
Comment (pmauduit) [inline console/src/main/webapp/WEB-INF/spring/webmvc-config.xml:327]: false by default ?

**Review by pmauduit — CHANGES_REQUESTED** [review]

File: console/src/main/webapp/WEB-INF/spring/webmvc-config.xml (hunk 5 above)
Comment (marwanehcine) [reply console/src/main/webapp/WEB-INF/spring/webmvc-config.xml]: done.

File: console/src/main/java/org/georchestra/console/events/DefaultPropsInitializer.java (hunk 4 above)
Comment (pmauduit) [reply console/src/main/java/org/georchestra/console/events/DefaultPropsInitializer.java]: you can still put the exception as argument from the Logger.error call I'd expect.

**Pull request merged**